src/infill/SpaghettiInfill.cpp

```diff
@@ -61,38 +65,38 @@ void SpaghettiInfill::generateTotalSpaghettiInfill(SliceMeshStorage& mesh)
 
 void SpaghettiInfill::generateSpaghettiInfill(SliceMeshStorage& mesh)
 {
-    const bool total_volume_at_once = !mesh.getSettingBoolean("spaghetti_infill_stepped");
+    const bool total_volume_at_once = !mesh.settings.get<bool>("spaghetti_infill_stepped");
 
     if (total_volume_at_once)
     {
         generateTotalSpaghettiInfill(mesh);
         return;
     }
 
-    const coord_t line_width = mesh.getSettingInMicrons("infill_line_width");
-    coord_t spaghetti_max_height = mesh.getSettingInMicrons("spaghetti_max_height");
+    const coord_t line_width = mesh.settings.get<coord_t>("infill_line_width");
+    coord_t spaghetti_max_height = mesh.settings.get<coord_t>("spaghetti_max_height");
 
-    coord_t filling_area_inset = mesh.getSettingInMicrons("spaghetti_inset");
+    coord_t filling_area_inset = mesh.settings.get<coord_t>("spaghetti_inset");
     
     coord_t connection_inset_dist;
-    if (mesh.getSettingInAngleDegrees("spaghetti_max_infill_angle") >= 90)
+    if (mesh.settings.get<AngleDegrees>("spaghetti_max_infill_angle") >= 90)
     {
         connection_inset_dist = MM2INT(500); // big enough for all standard printers.
     }
     else
     {
-        connection_inset_dist = tan(mesh.getSettingInAngleRadians("spaghetti_max_infill_angle")) * mesh.getSettingInMicrons("layer_height"); // Horizontal component of the spaghetti_max_infill_angle
+        connection_inset_dist = tan(mesh.settings.get<AngleRadians>("spaghetti_max_infill_angle")) * mesh.settings.get<coord_t>("layer_height"); // Horizontal component of the spaghetti_max_infill_angle
     }
 
     std::list<SpaghettiInfill::InfillPillar> pillar_base;
     coord_t current_z = 0;
 
-    size_t max_layer = mesh.layers.size() - 1 - mesh.getSettingAsCount("top_layers");
-    for (size_t layer_idx = 0; layer_idx <= max_layer; layer_idx++) //Skip every few layers, but extrude more.
+    LayerIndex max_layer = mesh.layers.size() - 1 - mesh.settings.get<size_t>("top_layers");
+    for (LayerIndex layer_idx = 0; layer_idx <= max_layer; layer_idx++) //Skip every few layers, but extrude more.
     {
-        const coord_t layer_height = (layer_idx == 0)? mesh.getSettingInMicrons("layer_height_0") : mesh.getSettingInMicrons("layer_height");
+        const coord_t layer_height = (layer_idx == 0)? mesh.settings.get<coord_t>("layer_height_0") : mesh.settings.get<coord_t>("layer_height");
         current_z += layer_height;
-        if (static_cast<int>(layer_idx) < mesh.getSettingAsCount("bottom_layers"))
+        if (layer_idx < static_cast<LayerIndex>(mesh.settings.get<size_t>("bottom_layers")))
```

Comment:
Because the setting is a number of layers (not a layer index), and then gets interpreted as "when you're this many layers above the build plate then do blabla". Layer indices are in the front-end displayed as 1-indexed, starting counting from 1. In the back-end we need to interpret this as 0-indexed, so if you were to get this as a layer index, you'd subtract 1 from the number of bottom layers which is not what you want.